main/zoe/guide/offer-safety.md

```diff
@@ -2,66 +2,61 @@
+using keywords. In this example, I would send Zoe a $100 payment to
+escrow my offer, as Zoe expects to be able to immediately escrow the `give:`-specified payment.
 
-After the user escrows, the user gets a promise for a payout from Zoe.
-It is this payout that offer safety is enforced over. Importantly, the
-payout either has to be what the user wanted, or it has to be a full
-refund.
+After the user escrows, they get a promise for a payout from Zoe.
+This is the payout that offer safety is enforced over. The payout *must*
+be either what the user wanted (an event ticket in the above example)
+or a full refund of what they escrowed ($100 in this example).
 
-We are able to enforce offer safety because the payout is under the
-control of Zoe. In the above example, if I am trying to buy my event
-ticket using a smart contract on Zoe, that smart contract can tell Zoe
-to update its bookkeeping to say that I've successfully bought a
-ticket. But, Zoe will only actually update its records and give me a
-payout of an event ticket if the update is offer-safe and conserves
-total supply.
+We can enforce offer safety because Zoe controls the payout. In the example, 
+if I try to buy my event ticket using a smart contract on Zoe, the contract
+can tell Zoe to update its bookkeeping to say I've bought a ticket.
+But, Zoe only actually updates its records and gives me an event ticket
+payout if that update is offer-safe and conserves total supply.
 
-The code for enforcing offer safety can be found in
-[offerSafety.js](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/offerSafety.js), and tests, including
-edge cases, can be found in [test-offerSafety.js](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/test/unitTests/test-offerSafety.js).
+The offer safety enforcement code is in
+[offerSafety.js](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contractFacet/offerSafety.js).
+Tests, including edge cases, are in [test-offerSafety.js](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/test/unitTests/test-offerSafety.js).
 
-## Offer Safety Gotchas
+## Implementation Questions
 
-### Can I get a full refund *and* what I wanted under offer safety?
+**Under offer safety, can I get a full refund *and* what I wanted?**
 
-Yes, under offer safety you can get a full refund *and* get what you
-wanted. Offer safety guarantees that at least one of these is true.
-Both could be true.
+Yes, you can get a full refund *and* what you wanted. Offer safety guarantees at least 
+one is true. Both could also be true.
 
-### What if there are no rules under `give`?
+**What if there are no rules under `give`?**
 
-If there are no rules under `give` or `give` is omitted, then
-trivially, whatever you get will fulfill offer-safety, because you are
-always getting a full refund of what you put in, which was nothing.
+If there are no rules or `give` is omitted, then you get a full refund
```

Comment:
This sounds like they won't get what they wanted. Can we state that they may also get what they wanted in addition to the refund in this case?

Reply:
We've thought of several examples where a contract would expect to payout for no explicit payment. e.g. a coat check or a drop-box would commit to giving the proceeds to whomever has the invitation.

Reply:
I'm not sure I follow. Say for coat check, do you mean the invitation is the equivalent of a numbered ticket indicating which slot my coat is in, because I'd think the ticket would an asset be what I gave to get my coat (i.e. the right to whatever is in slot 56)?.